### DISCUSSION TITLE
Language unknown before ocr

### DISCUSSION THREAD

**Discussion by sasumay** [PaddlePaddle/PaddleOCR#6700] [Q&A] [answered]
I have image files for which I don't know the languages in them. I cannot call paddleocr 20+ times to figure out the language. 
Can you please help with any ideas or suggestions using capabilities of paddleocr?

**Comment by Evezerest** [comment, marked as answer]
Looks like you need a language classifier between text detection and text recognition. We don't currently have such an algorithm. Maybe you can look at some common classification algorithms.
Could you explain more about your requirement? (under what circumstances do you need to recognize 20+ languages?)

**Reply by sasumay** [reply]
Thank you for the response. 
My requirement is that if I get an image that is not in English, it needs to be translated to English. The input image can be in any language, and to get the right text, I need to determine the language before I call paddleocr.
I couldn't find any opensource language classifier from images but I will keep looking.

**Reply by cweiler-blumatix** [reply]
Any luck yet - looking also for language classifier.

**Reply by Evezerest** [reply]
YES, recently PaddleClas open-sourced a language classification model. Click [here](https://github.com/PaddlePaddle/PaddleClas/blob/release/2.4/docs/en/PULC/PULC_language_classification_en.md) for more information

**Reply by NaumanHSA** [reply]
Thank you guys for sharing this.

**Reply by ariefwijaya** [reply]
@Evezerest Hi, thanks for reference, So for the case of a document, the idea is we need to infer all the text and get a list of languages ​​available on the page or we can infer all the text at once?
